airqo_monitor/external/thingspeak.py

```diff
@@ -61,21 +65,26 @@ def get_data_for_channel(channel, start_time=None, end_time=None):
     return all_data
 
 
-def get_channel_ids_to_names():
-    """
-    Get all relevant channels to this tool (channels with AIRQO and without INACTIVE in the name)
-    """
+def get_all_channels():
     api_key = os.environ.get('THINGSPEAK_USER_API_KEY')
     full_url = '{}/?api_key={}'.format(THINGSPEAK_CHANNELS_LIST_URL, api_key)
     channels = make_get_call(full_url)
+    return channels
+
 
-    channel_ids_to_names = defaultdict(dict)
-    for channel in channels:
-        name = channel['name']
-        if AIR_QUALITY_MONITOR_KEYWORD in name and INACTIVE_MONITOR_KEYWORD not in name:
-            channel_ids_to_names[channel['id']]["name"] = name
+def get_all_channels_cached():
+    cached_value = cache.get('get-all-channels')
+    if cached_value is None:
+        cached_value = get_all_channels()
+        cache.set('get-all-channels', cached_value, timeout=30 * 60)
+    return cached_value
 
-    return channel_ids_to_names
+
+def get_all_channels_by_type(channel_type):
```

Comment:
no caching because we only get a specific type right now when we update?

Reply:
yeah before i had it cached because we were going to make the exact same call a bunch of times, so we didn't actually need to keep repeating. For this we're only actually calling it when we need the new data.